crates/ruff_workspace/src/resolver.rs

```diff
@@ -140,19 +144,31 @@ impl<'a> Resolver<'a> {
     }
 
     /// Add a resolved [`Settings`] under a given [`PathBuf`] scope.
-    fn add(&mut self, path: PathBuf, settings: Settings) {
-        self.settings.insert(path, settings);
+    fn add(&mut self, path: &Path, settings: Settings) {
+        self.settings.push(settings);
+
+        // normalize the path to use `/` separators and escape the '{' and '}' characters,
+        // which matchit uses for routing parameters
```

Comment:
Are there any other characters that we need to watch out for here? In my previous PR I mentioned colons, which seems irrelevant, but e.g. `{*foo}` is a special routing parameter in matchit, right? I assume that's _also_ irrelevant as long as we're escaping the `{` and `}` characters.

Reply:
`*` is only considered after an unescaped `{`, so that shouldn't be a problem.